### PR TITLE
Class4 piglatin

### PR DESCRIPTION
**Pull request by nathanhbsimmons**
Function plan is included above pigLatin() function in comments

### CODE REVIEW

File: 02week/pigLatin.js [hunk 1]
```diff
@@ -2,22 +2,59 @@
 
 const assert = require('assert');
 const readline = require('readline');
-const rl = readline.createInterface({
-  input: process.stdin,
-  output: process.stdout
-});
+const rl = readline.createInterface({input: process.stdin, output: process.stdout});
 
+// Pig Latin App Plan:
```
Comment (HippErger) [inline 02week/pigLatin.js:7]: Love your plan!!

**Review by reneemeyer — APPROVED** [review]
Great job! Amazing plan and function plan

File: 02week/pigLatin.js [hunk 2]
```diff
@@ -2,22 +2,59 @@
...
+// -isInputValid() = checks for string containing only letters as valid input (match method), if valid run checkStartingLetter()
+//
+// -checkStartingLetter() = checks if word starts with vowel or consonant and formats accordingly
+// (vowelIndex = 0 then add ‘yay’ to end of string, else run rearrangeLetters())
+//
+// -rearrangeLetters() = moves letters before first vowel to end of string and adds ‘ay’ (substring method)
 
 function pigLatin(word) {
+  const formattedWord = word.toLowerCase().trim()
+  const firstVowel = formattedWord.match(/[aeiou]/)
```
Comment (HippErger) [inline 02week/pigLatin.js:29]: This is a really slick way of setting the vowel Index!

File: 02week/pigLatin.js [hunk 3]
```diff
@@ -2,22 +2,59 @@
...
+    } else {
+      return rearrangeLetters()
+    }
+  }
+
+  const rearrangeLetters = () => {
+    return `${formattedWord.substring(vowelIndex)}${formattedWord.substring(0, (vowelIndex))}ay`
+  }
 
+  return isInputValid()
```
Comment (HippErger) [inline 02week/pigLatin.js:52]: In the future its best practice to have all of your functions sitting outside of each other being ready to be called if need be.

**Pull request merged**